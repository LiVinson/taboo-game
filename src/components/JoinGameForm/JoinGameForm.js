import React from 'react'
import PropTypes from 'prop-types'
import { Formik, Field, ErrorMessage } from 'formik'
import { Form, FormSection, TextInput, TextLabel, ErrorText } from 'components/shared/FormElements'

const validate = (values) => {
	const errors = {}
	if (!values.name) {
		errors.name = 'Name is Required'
	} else if (values.name.length > 25) {
		errors.name = 'Must be 25 characters or less'
	}

	if (!values.gamecode) {
		errors.gamecode = 'Game code is required'
	}
	return errors
}

const JoinGameForm = (props) => {
	return (
		<Formik
			initialValues={props.initialValues}
			validate={validate}
			onSubmit={(values, { setSubmitting }) => props.handleSubmit(values, setSubmitting)}
		>
			{(formik) => (
				<Form id="joinGameForm">
					<FormSection>
						<TextLabel htmlFor="name">Enter Your Name</TextLabel>
						<Field id="name" name="name" type="text" as={TextInput} placeholder="" tabIndex="1" />
						<ErrorMessage name="name" component={ErrorText} />
					</FormSection>
					<FormSection>
						<TextLabel htmlFor="gamecode">Enter a Game Code</TextLabel>
						<Field id="gamecode" name="gamecode" type="text" as={TextInput} placeholder="" tabIndex="1" />
						<ErrorMessage name="gamecode" component={ErrorText} />
					</FormSection>
				</Form>
			)}
		</Formik>
	)
}

JoinGameForm.propTypes = {
	initialValues: PropTypes.object.isRequired,
}

export default JoinGameForm
