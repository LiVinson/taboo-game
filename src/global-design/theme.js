const theme = {
  color: {
    primary: "#1982C4", //light blue
    accent1: "#8AC926", //lime green
    accent2: "#ee5b2c", //orange
    accent3: "#a23b72", //purple
    primaryLight: "#58b1e9", //light-blue
    white: "#fff",
    offWhite: "#f7f9f9",
    lightGray1: "#f1f1f1",
    lightGray2: "#ebebeb",
    grayDark1: "#757575",
    grayDark2: "#333333"
  },
  text: {
    display: "'Oswald', sans-serif",
  },
  fontSize: {
    xsmall: "1.8rem",
    small: "2.2rem",
    medium: "2.7rem",
    large: "4rem",
    xlarge: "7rem",
  },
}


export default theme
