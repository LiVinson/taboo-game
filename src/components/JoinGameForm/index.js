import JoinGameForm  from "./JoinGameForm"

export default JoinGameForm