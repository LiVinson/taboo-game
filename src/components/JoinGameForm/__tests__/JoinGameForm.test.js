import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import theme from "../../../global-design/theme"
import JoinGameForm from "components/JoinGameForm"

test("JoinGameForm renders correctly", () => {
  const props = {
    handleSubmit: jest.fn(),
    initialValues: {
      name: "",
      gamecode: "remote",
    },
  }

  const wrapper = renderer.create(
    <ThemeProvider theme={theme}>
      <JoinGameForm {...props} />
    </ThemeProvider>
  )
  expect(wrapper).toMatchSnapshot()
})
