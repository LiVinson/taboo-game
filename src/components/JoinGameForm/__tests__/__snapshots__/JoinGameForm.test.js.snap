// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinGameForm renders correctly 1`] = `
<form
  action="#"
  className="sc-AxjAm hyBwHU"
  id="joinGameForm"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="sc-AxirZ itgPJw"
  >
    <label
      className="sc-Axmtr sc-AxmLO jMZnwH"
      htmlFor="name"
    >
      Enter Your Name
    </label>
    <input
      className="sc-AxhUy dcifbi"
      id="name"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="1"
      type="text"
      value=""
    />
  </div>
  <div
    className="sc-AxirZ itgPJw"
  >
    <label
      className="sc-Axmtr sc-AxmLO jMZnwH"
      htmlFor="gamecode"
    >
      Enter a Game Code
    </label>
    <input
      className="sc-AxhUy dcifbi"
      id="gamecode"
      name="gamecode"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder=""
      tabIndex="1"
      type="text"
      value="remote"
    />
  </div>
</form>
`;
